coltypes,coldata: remove Zero method

yuzefovich opened the pull request:
**coltypes: remove execgen.ZERO**

It is no longer used.

Release note: None

**coldata: remove Bytes.Zero method and panic on resetting a window**

This commit moves the implementation of Bytes.Zero into Bytes.Reset (the
only caller of the former). It also adds a panic in case Reset is called
on a window into Bytes.

Release note: None

**colexec: remove zerocolumns template**

Some of the zero<Type>Columns are no longer used, so those are removed.
The code itself is moved from a template file into a regular one.

Release note: None

yuzefovich committed:
coltypes: remove execgen.ZERO

It is no longer used.

Release note: None

yuzefovich committed:
coldata: remove Bytes.Zero method and panic on resetting a window

This commit moves the implementation of Bytes.Zero into Bytes.Reset (the
only caller of the former). It also adds a panic in case Reset is called
on a window into Bytes.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/43603)
<!-- Reviewable:end -->

asubiotto reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> but this unfortunately doesn't fix the memory corruption I was talking about.

> Reviewed 2 of 2 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43603#-:-LyVRfe9AeIbLewdIwdy:b-vzu6pj)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[pkg/col/coltypes/types.go, line 309 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/43603#-LyVRV0k7dg9QYR7mgVr:-LyVRV0k7dg9QYR7mgVs:b-nfgjvs) ([raw file](https://github.com/cockroachdb/cockroach/blob/8f946b8bb62629002c5e958373b81ca9e1920dd6/pkg/col/coltypes/types.go#L309)):*
> ```Go
> }`, target)
> 	}
> 	return fmt.Sprintf("for n := 0; n < len(%[1]s); n += copy(%[1]s[n:], zero%sColumn) {}", target, t.String())
> ```

Can we remove the `zero<Type>Column`s as well?


<!-- Sent from Reviewable.io -->

yuzefovich committed:
colexec: remove zerocolumns template

Some of the zero<Type>Columns are no longer used, so those are removed.
The code itself is moved from a template file into a regular one.

Release note: None

yuzefovich reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43603)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @asubiotto)

___
*[pkg/col/coltypes/types.go, line 309 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/43603#-LyVRV0k7dg9QYR7mgVr:-LyVdoQlEUbpb6IeW77R:bv9j3p4) ([raw file](https://github.com/cockroachdb/cockroach/blob/8f946b8bb62629002c5e958373b81ca9e1920dd6/pkg/col/coltypes/types.go#L309)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

Can we remove the `zero<Type>Column`s as well?
</blockquote></details>

Removed the template. Some of `zero<Type>Column`s are still used though.


<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
